tests/e2e/uninstall.go

```diff
@@ -0,0 +1,194 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// requireRemoveKoperatorCRDs deletes the koperator CRDs
+func requireRemoveKoperatorCRDs(kubectlOptions k8s.KubectlOptions) {
+	It("Removing koperator CRDs", func() {
+		for _, crd := range koperatorCRDs() {
+			deleteK8sResourceNoErrNotFound(kubectlOptions, defaultDeletionTimeout, crdKind, crd)
+		}
+	})
+}
+
+// requireUninstallingKoperator uninstall koperator Helm chart and removes Koperator's CRDs.
+func requireUninstallingKoperator(kubectlOptions k8s.KubectlOptions) {
+	When("Uninstalling Koperator", func() {
+		requireUninstallingKoperatorHelmChart(kubectlOptions)
+		requireRemoveKoperatorCRDs(kubectlOptions)
+	})
+}
+
+// requireUninstallingKoperatorHelmChart uninstall Koperator Helm chart
+// and checks the success of that operation.
+func requireUninstallingKoperatorHelmChart(kubectlOptions k8s.KubectlOptions) {
+	It("Uninstalling Koperator Helm chart", func() {
+		uninstallHelmChartIfExists(kubectlOptions, koperatorLocalHelmDescriptor.ReleaseName, true)
+		By("Verifying Koperator helm chart resources cleanup")
+		k8sResourceKinds, err := listK8sResourceKinds(kubectlOptions, "")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		koperatorAvailableResourceKinds := stringSlicesInstersect(koperatorCRDs(), k8sResourceKinds)
+		koperatorAvailableResourceKinds = append(koperatorAvailableResourceKinds, basicK8sResourceKinds()...)
+
+		remainedResources, err := getK8sResources(kubectlOptions,
+			koperatorAvailableResourceKinds,
+			fmt.Sprintf(managedByHelmLabelTemplate, koperatorLocalHelmDescriptor.ReleaseName),
+			"",
+			kubectlArgGoTemplateKindNameNamespace,
+			"--all-namespaces")
+
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(remainedResources).Should(BeEmpty())
+	})
+}
+
+// requireUninstallingZookeeperOperator uninstall Zookeeper-operator Helm chart
+// and remove CRDs.
+func requireUninstallingZookeeperOperator(kubectlOptions k8s.KubectlOptions) {
+	When("Uninstalling zookeeper-operator", func() {
+		requireUninstallingZookeeperOperatorHelmChart(kubectlOptions)
+		requireRemoveZookeeperOperatorCRDs(kubectlOptions)
+	})
+}
+
+// requireUninstallingZookeeperOperatorHelmChart uninstall Zookeeper-operator Helm chart
+// and checks the success of that operation.
+func requireUninstallingZookeeperOperatorHelmChart(kubectlOptions k8s.KubectlOptions) {
+	It("Uninstalling zookeeper-operator Helm chart", func() {
+		uninstallHelmChartIfExists(kubectlOptions, zookeeperOperatorHelmDescriptor.ReleaseName, true)
```

Comment:
**Required**: We **MUST** check potential errors here.

```suggestion
		err := uninstallHelmChartIfExists(kubectlOptions, zookeeperOperatorHelmDescriptor.ReleaseName, true)
		
		Expect(err).NotTo(HaveOccurred())
```

Reply:
Fixed in [63ed2cd](https://github.com/banzaicloud/koperator/pull/987/commits/63ed2cdca59f233e329f912c042c39a21ecf3a7c).